opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing
+from enum import Enum
+
+from .. import Span, SpanProcessor
+
+
+class ExportResult(Enum):
+    SUCCESS = 0
+    FAILED_RETRYABLE = 1
+    FAILED_NONE_RETRYABLE = 2
+
+
+class Exporter:
+    """Interface that allows different services to export recorded spans in
+    its own format.
+
+    To export data this MUST be registered to the :class`..Tracer` using a
+    `SimpleSpanProcessor` or a `BatchSpanProcessor`.
+    """
+
+    def export(self, spans: typing.Sequence[Span]) -> "ExportResult":
+        """Exports a batch of telemetry data.
+
+        Args:
+            spans: The list of `Span`s to be exported
+
+        Returns:
+            The result of the export
+        """
+
+    def shutdown(self) -> None:
+        """Shuts down the exporter.
+
+        Called when the SDK is shut down.
+        """
+
+
+class SimpleSpanProcessor(SpanProcessor):
+    """An implementation of `SpanProcessor` that passes ended spans directly
+    to the configured `Exporter`.
+    """
+
+    def __init__(self, exporter: Exporter):
+        self._exporter = exporter
+
+    def on_start(self, span: Span) -> None:
+        pass  # do nothing
+
+    def on_end(self, span: Span) -> None:
+        try:
+            self._exporter.export((span,))
+        except BaseException as exc:
+            # TODO: how to log?
+            print("Exception while exporting data: {}".format(exc))
```

Comment:
Probably we should use the `logging` module for such messages (I think OpenCensus did that). In any case, we must at least not print to stdout.

Reply:
I'd just log to stderr while we decide if logging should be use or not.

Reply:
I believe we should follow [Python logging practice](https://docs.python.org/3/howto/logging.html#logging-advanced-tutorial).

Reply:
You're right, I did it.